pkg/estimator/server/framework/plugins/resourcequota/resourcequota.go

```diff
@@ -0,0 +1,361 @@
+		klog.V(5).Info("Estimator Plugin", "name", Name, "enabled", pl.enabled)
+		return replica, framework.NewResult(framework.Noopperation, fmt.Sprintf("%s is disabled", pl.Name()))
+	}
+	namespace := replicaRequirements.Namespace
+	priorityClassName := replicaRequirements.PriorityClassName
+
+	rqList, err := pl.rqLister.ResourceQuotas(namespace).List(labels.Everything())
+	if err != nil {
+		klog.Error(err, "fail to list resource quota", "namespace", namespace)
+		return replica, framework.AsResult(err)
+	}
+	for _, rq := range rqList {
+		rqEvaluator := newResourceQuotaEvaluator(rq, priorityClassName)
+		replicaFromRqEvaluator := rqEvaluator.evaluate(replicaRequirements)
+		if replicaFromRqEvaluator < replica {
+			replica = replicaFromRqEvaluator
+		}
+	}
+
+	var result *framework.Result
+	if replica == math.MaxInt32 {
+		result = framework.NewResult(framework.Noopperation, fmt.Sprintf("%s has no operation on input replicaRequirements", pl.Name()))
+	} else if replica == 0 {
+		result = framework.NewResult(framework.Unschedulable, fmt.Sprintf("zero replica is estimated by %s", pl.Name()))
+	} else {
+		result = framework.NewResult(framework.Success)
+	}
+	return replica, result
+}
+
+type resourceQuotaEvaluator struct {
+	//key (string) is the name of the resource quota
+	//value (ResourceList) is the free resources calculated by (hard - used) from the resource quota status
+	resourceRequest map[string]corev1.ResourceList
+}
+
+func newResourceQuotaEvaluator(rq *corev1.ResourceQuota, priorityClassName string) *resourceQuotaEvaluator {
+	selectors := getScopeSelectorsFromQuota(rq)
+	resources := make(map[string]corev1.ResourceList)
+
+	for _, selector := range selectors {
+		matchScope, err := matchesScope(selector, priorityClassName)
+		if err != nil {
+			klog.Error(err, "matchesScope failed")
+			continue
+		}
+		if !matchScope {
+			continue
+		}
+		matchResource := matchingResources(resourceNames(rq.Status.Hard))
+		if len(matchResource) != 0 {
+			resource := calculateFreeResources(rq, matchResource)
+			resources[rq.Name] = resource
+			break
+		}
+	}
+	return &resourceQuotaEvaluator{
+		resourceRequest: resources,
+	}
+}
+
+func (e *resourceQuotaEvaluator) evaluate(replicaRequirements *pb.ReplicaRequirements) int32 {
+	var result int32 = math.MaxInt32
+	for _, resourceList := range e.resourceRequest {
+		filteredRequiredResourceList := corev1.ResourceList{}
+		// If the resource in pb.ReplicaRequirements is not the ResourceQuota ResourceList, we skip it.
+		for resourceName, request := range replicaRequirements.ResourceRequest {
+			if _, ok := resourceList[resourceName]; ok {
+				filteredRequiredResourceList[resourceName] = request
+			}
+		}
+		resource := util.NewResource(resourceList)
+		resource.AllowedPodNumber = math.MaxInt64
+		allowed := resource.MaxDivided(filteredRequiredResourceList)
+		// continue the loop to avoid integer overflow
+		if allowed == math.MaxInt64 {
+			continue
+		}
+		replica := int32(allowed)
```

Comment:
Is the `allowed` possible to exceed `math.MaxInt32`, the converted value might not be expected?

Reply:
fix it according to the comment.